devices/Sim7080/Sim7080G.cs

```diff
@@ -0,0 +1,301 @@
+{
+    /// <summary>
+    /// Sim7080G Multi-Band CAT-M(eMTC) and NB-IoT module
+    /// </summary>
+    public class Sim7080G
+    {
+        /// <summary>
+        /// The UART <see cref="SerialPort"/> for communication with the modem
+        /// </summary>
+        private readonly SerialPort _serialPort;
+
+        /// <summary>
+        /// Constructor
+        /// </summary>
+        /// <param name="serialPort">The UART <see cref="SerialPort"/> for communication with the modem</param>
+        public Sim7080G(SerialPort serialPort) => _serialPort = serialPort;
+
+        /// <summary>
+        /// Number of retries on failure
+        /// </summary>
+        public int Retry { get; set; } = 3;
+
+        /// <summary>
+        /// The network system mode for wireless and cellular network communication service
+        /// </summary>
+        public SystemMode SystemMode { get; private set; }
+
+        /// <summary>
+        /// Cellular network connection status
+        /// </summary>
+        public ConnectionStatus NetworkConnected { get; set; } = ConnectionStatus.Disconnected;
+
+        /// <summary>
+        /// MQTT endpoint connection status
+        /// </summary>
+        public ConnectionStatus EndpointConnected { get; set; } = ConnectionStatus.Disconnected;
+
+        /// <summary>
+        /// MQTT subscription topic connection status
+        /// </summary>
+        public ConnectionStatus TopicConnected { get; set; } = ConnectionStatus.Disconnected;
+
+        /// <summary>
+        /// Cellular Network Operator
+        /// </summary>
+        public string Operator { get; private set; } = "Unknown";
+
+        /// <summary>
+        /// The Public IP Address
+        /// </summary>
+        public string IPAddress { get; private set; } = "0.0.0.0";
+
+        /// <summary>
+        /// Cloud-2-Device subscription topic Name
+        /// </summary>
+        public string SubTopic { get; set; }
+
+        /// <summary>
+        /// Device-2-Cloud publication topic Name
+        /// </summary>
+        public string PubTopic { get; set; }
+
+        /// <summary>
+        /// Set the network system mode for wireless and cellular network communication service
+        /// </summary>
+        /// <param name="systemMode">The desired <see cref="SystemMode"/></param>
+        /// <param name="enableReporting">Report the network system mode information</param>
+        /// <param name="wait">The time to wait to switch system mode</param>
+        public void SetNetworkSystemMode(SystemMode systemMode = SystemMode.GSM, bool enableReporting = true, int wait = 5000)
+        {
+            SimController.SetSystemMode(_serialPort, systemMode, enableReporting);
+
+            Thread.Sleep(wait);
+        }
+
+        /// <summary>
+        /// Connect to the cellular network
+        /// </summary>
+        //// <param name="apn">Cellular network access point name</param>
+        public void NetworkConnect(string apn)
```

Comment:
doesn't it make sense to return either the enum either a Boolean to check the status without having to explicitly call the enum after?

Reply:
We could that, but then I would do this as an addition. Sometimes you would like to check the status without having to call the method again.

Reply:
Absolutely. I was thinking of exposing the underlying enum that you're using internally

Reply:
I changed the return status to the ConnectionStatus Enum Value; the one I am using internally